plugin-grace-hopper/src/probe.rs

```diff
@@ -0,0 +1,88 @@
+use std::{
+    fs::File,
+    io::{Read, Seek},
+    path::Path,
+};
+
+use alumet::{
+    measurement::MeasurementPoint,
+    metrics::TypedMetricId,
+    pipeline::Source,
+    plugin::AlumetPluginStart,
+    resources::{Resource, ResourceConsumer},
+    units::{PrefixedUnit, Unit},
+};
+
+pub struct GraceHopperProbe {
+    socket: String,
+    kind: String,
+    file: Option<File>,
+    consumer: ResourceConsumer,
+    metric: Option<TypedMetricId<u64>>,
+}
+
+impl GraceHopperProbe {
+    pub fn new(
+        alumet: &mut AlumetPluginStart,
+        socket: String,
+        kind: String,
+        filepath: Option<&Path>,
+    ) -> Result<Self, anyhow::Error> {
+        let sensor_kind = kind.to_lowercase();
+        let metric = alumet
+            .create_metric::<u64>(
+                "consumption",
+                PrefixedUnit::micro(Unit::Watt),
+                "Power consumption of the sensor",
+            )
+            .ok();
+        let mut probe: GraceHopperProbe = GraceHopperProbe {
+            socket,
+            kind: sensor_kind,
+            file: None,
+            metric,
+            consumer: ResourceConsumer::LocalMachine,
+        };
+
+        if let Some(filepath) = filepath {
+            probe.file = Some(File::open(filepath.join("power1_average"))?);
+        }
+        Ok(probe)
+    }
+}
+
+impl Source for GraceHopperProbe {
+    fn poll(
+        &mut self,
+        measurements: &mut alumet::measurement::MeasurementAccumulator,
+        timestamp: alumet::measurement::Timestamp,
+    ) -> Result<(), alumet::pipeline::elements::error::PollError> {
+        let mut buffer = String::new();
+        if let Some(file) = &mut self.file {
```

Comment:
I don't think this check make sense here, no file == no source. So a source should always have a file (remove option).